financial-connections/src/main/java/com/stripe/android/financialconnections/features/partnerauth/PartnerAuthViewModel.kt

```diff
@@ -293,6 +300,34 @@ internal class PartnerAuthViewModel @Inject constructor(
         }
     }
 
+    /**
+     * Cancels the given [authSession] and navigates to the next pane (non-OAuth) / retries (OAuth).
+     */
+    private suspend fun cancelAuthSessionAndContinue(
+        authSession: FinancialConnectionsAuthorizationSession
+    ) {
+        val result = cancelAuthorizationSession(authSession.id)
+        if (authSession.isOAuth) {
+            // For OAuth institutions, create a new session and navigate to its nextPane (prepane).
+            logger.debug("Creating a new session for this OAuth institution")
+            // Send retry event as we're presenting the prepane again.
+            postAuthSessionEvent(authSession.id, AuthSessionEvent.Retry(Date()))
+            // for OAuth institutions, we remain on the pre-pane,
+            // but create a brand new auth session
+            setState { copy(authenticationStatus = Uninitialized) }
+            createAuthSession()
+        } else {
+            // For OAuth institutions, navigate to Session cancellation's next pane.
```

Comment:
Should this be `For non-OAuth institutions ....` ?

Reply:
https://github.com/stripe/stripe-android/pull/7125/commits/4da41b114a1ca07d40391d36f9ca93f3510e266f